### DISCUSSION TITLE
java udf  引入第三方依赖

### DISCUSSION THREAD

**Discussion by foundation1** [apache/doris#32155] [A - General / Q&A]
自己写的 java udf  打包应该用什么插件才可以把 第三方 依赖打入 并且正常执行

**Comment by foundation1** [comment]
使用官网示例中的 打包插件  打包后  有两个jar包 一个是包含所有依赖的  一个是不包含的   但是都无法使用

**Reply by ihadoop** [reply]
can you provide the  error message?or more detail messages?

**Reply by foundation1** [reply]
这个是我打包的jar 一个是包含依赖jar包一个没有包含
![image](https://github.com/apache/doris/assets/28500506/d44f12ab-3b2c-4604-8627-d2396e2eeed4)
其中 jvs-doris-udf-jar-with-dependencies.jar里面的内容是 
![image](https://github.com/apache/doris/assets/28500506/b9d68be6-b6e1-4e60-8fe9-8c4e026c5d43)

执行 自定义 函数 提示 没有找到对应的 class
![image](https://github.com/apache/doris/assets/28500506/7c2a6780-bdb7-40e6-bb41-ca3bd6158981)
![image](https://github.com/apache/doris/assets/28500506/01bbb4dd-8657-4aef-bcce-2b7b2ed4f2e6)
以下是我的  函数 内容
![image](https://github.com/apache/doris/assets/28500506/c7b141db-485b-42d3-8d71-87c048fb035f)
doris 中声明如下:
{"symbol":"cn.bctools.doris.function.core.user.GetUserInfoFunction","object_file":"http://10.0.0.200:9000/jvs-public/java-udf-demo-jar-with-dependencies.jar","md5":"317e7dc39cb23e3f42674eb00e6b1ac2"}

是不是应该使用其他的 打包插件才行   主要是 StrUtil这个第三方包他可以访问到  就很奇怪

**Reply by ihadoop** [reply]
your udf is right，but the depend jar “mysql driver” is not contained is the classpath。you can add driver jar into your  final package jar

**Reply by foundation1** [reply]
但是  实际是有的
![image](https://github.com/apache/doris/assets/28500506/28b15abb-3e12-47b1-9a37-1c9689bbb56c)
我在想是不是打包 插件 需要更换
